lib/pluginmanager/sources/local.rb

```diff
@@ -0,0 +1,26 @@
+# encoding: utf-8
+require "rubygems/package"
+require "pluginmanager/sources/base"
+require "fileutils"
+
+module LogStash::PluginManager::Sources
+
+  class Local < Base
+
+    def exist?
+      ::File.exist?("#{uri}")
+    end
+
+    def fetch(dest="")
+      base_dir = (dest.empty? ? LogStash::Environment::LOGSTASH_HOME : dest )
+      current_path = ::File.join(base_dir, ::File.basename("#{uri}"))
```

Comment:
yes, ^^ is the reason why I used to_s to convert it ... you should use ``file://` to be detected as local uri, otherwise anything without schema is generic.